packages/type-utils/tests/getDeclaration.test.ts

```diff
@@ -0,0 +1,47 @@
+import type { TSESTree } from '@typescript-eslint/types';
+import type { ParserServicesWithTypeInformation } from '@typescript-eslint/typescript-estree';
+import type * as ts from 'typescript';
+
+import { getDeclaration } from '../src';
+
+const node = {} as TSESTree.Node;
+
+describe('getDeclaration', () => {
+  describe('when symbol does not exist', () => {
+    it('returns null', () => {
+      const services = {
+        getSymbolAtLocation: () => undefined,
+      } as unknown as ParserServicesWithTypeInformation;
```

Comment:
Lots of `as unknown as` in this file. Duplicate code, and kind of hard to read through. Request: can you try extracting these to either shared variables or reusable helpers so that the individual functions don't have so many assertions?

Reply:
Thanks for your review 🙏 
Done https://github.com/typescript-eslint/typescript-eslint/pull/10074/commits/14db76cdf70aa00697a60dc0c08e79b327b497da